src/fragments/lib/datastore/native_common/advanced-workflows.mdx

```diff
@@ -0,0 +1,272 @@
+## Custom Primary Keys
+
+By default, all DataStore models have an `id` field by default that is automatically populated on the client with a UUID v4, allowing DataStore to generate non-colliding globally unique identifiers in a scalable way.
+
+While UUIDs have desirable properties (they are large, non-sequential and opaque), there are times when a custom primary key is needed. For instance, to:
+
+
+* Have friendly/readable identifiers (surrogate/opaque vs. natural keys)
+* Define composite primary keys
+* Customize data partitioning to optimize for scale (especially important when planning to handle large amounts of data in short periods of time)
+* Selectively synchronize data to clients (e.g. by fields like `deviceId`, `userId` or similar)
+* Prioritize the sort order in which objects are returned by the sync queries
+* Make existing data consumable and syncable by DataStore
+
+### Schema
+
+A schema with the typical `id` field looks like this:
+
+```graphql
+type Todo4 @model {
+	id: ID!
+	name: String!
+	description: String
+}
+```
+
+With custom primary keys, you can provide a unique value using the `@primaryKey` directive:
+
+```graphql
+type Todo @model {
+	customId: ID! @primaryKey
+	name: String!
+	description: String
+}
+```
+
+With sort key:
+
+```graphql
+type Todo @model {
+	customId: ID! @primaryKey(sortKeyFields: ["createdAt"])
+	name: String!
+	description: String
+	createdAt: AWSDateTime!
+	updatedAt: AWSDateTime
+}
+```
+
+With name and query field:
+
+```graphql
+# Custom PK + sort key w/ name + query field
+type Todo @model {
+	id: ID!
+	customId: ID!
+		@index(
+			name: "byCustomIdAndCreated"
+			sortKeyFields: ["createdAt"]
+			queryField: "todoByCustomIdAndCreatedAt"
+		)
+	name: String!
+	description: String
+	createdAt: AWSDateTime!
+	updatedAt: AWSDateTime
+}
+```
+
+### Creating records:
```

Comment:
I initially had all of the headings end in "with custom primary keys", but it seemed redundant in the right navigation panel (since all headings are displayed there), so I removed that. I think we should stay consistent across all headings, so if there is value in calling this out explicitly, we should make the update to all the headings (create, delete, query). My current thought is to exclude these - what do you think?

Reply:
See comment above on the "schemas"

Reply:
Discussed offline, will favor indexing on improved search results vs. redundancy of nav headers